utils/model_runners/gpt_model_runner.py

```diff
@@ -38,12 +44,46 @@ def predict(self, prompt: str) -> Tuple[Optional[str], Optional[float]]:
             "presence_penalty": 0,
             "frequency_penalty": 0
         })
-        
+
         headers = {
             "Content-Type": "application/json",
             "Accept": "application/json",
             "Authorization": "Bearer " + self.config.api_key
         }
-        
+        stime = datetime.now(timezone.utc)
+
         response = requests.request("POST", self.url, headers=headers, data=payload)
-        return json.loads(response.text)["choices"][0]["message"]["content"], None
\ No newline at end of file
+        delta =  datetime.now(timezone.utc) - stime
+        processing_time = delta.total_seconds() * 1000
+
+
+        response = json.loads(response.text)
+        output = response["choices"][0]["message"]["content"]
+        input_token_count = int(response["usage"]["prompt_tokens"])        
+        output_token_count = int(response["usage"]["completion_tokens"])
+        
```

Comment:
DRY - define this once as `usage_filename = self._metrics_folder + f"/{self._model_key}_usage.jsonl` and reuse